File: lv1_module2/ros2_ws/src/turtle_py/turtle_py/distance_watcher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import Float32


class DistanceWatcher(Node):

    def __init__(self):
        super().__init__('distance_watcher')

        self.declare_parameter('threshold', 15.0) #임계값 3.0으로 설정, 임계값 변경 필요시 숫자 변경

        self._sub = self.create_subscription(
            Float32, '/turtle_distance', self._distance_callback, 10) #10Hz로 발행

    def _distance_callback(self, msg: Float32) -> None:
        threshold = self.get_parameter('threshold').value
        if msg.data > threshold:
            self.get_logger().warn(
                f'거리 {msg.data:.3f} m가 임계값 {threshold:.3f} m를 초과했습니다') #3.0초과시 경고문구 출력


def main(args=None):
    rclpy.init(args=args)
    node = DistanceWatcher()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()

File: lv1_module2/ros2_ws/src/turtle_cpp/src/distance_watcher.cpp
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float32.hpp"

using std::placeholders::_1;

class DistanceWatcher : public rclcpp::Node {
 public:
  DistanceWatcher() : Node("distance_watcher") {
    this->declare_parameter<double>("threshold", 3.0);

    sub_ = this->create_subscription<std_msgs::msg::Float32>(
        "/turtle_distance", 10,
        std::bind(&DistanceWatcher::distanceCallback, this, _1));
  }

 private:
  void distanceCallback(const std_msgs::msg::Float32::SharedPtr msg) {
    const double threshold = this->get_parameter("threshold").as_double();
    if (msg->data > threshold) {
      RCLCPP_WARN(this->get_logger(), "거리 %.3f m가 임계값 %.3f m를 초과했습니다",
                  msg->data, threshold);
    }
  }

  rclcpp::Subscription<std_msgs::msg::Float32>::SharedPtr sub_;
};

int main(int argc, char** argv) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<DistanceWatcher>());
  rclcpp::shutdown();
  return 0;
}
